tree/ntuple/v7/src/RFieldVisitor.cxx

```diff
@@ -94,10 +98,195 @@ void ROOT::Experimental::RPrepareVisitor::VisitField(const Detail::RFieldBase &/
       fDeepestLevel = level;
 }
 
+//------------------------ RValueVisitor --------------------------------
+
+void ROOT::Experimental::RValueVisitor::VisitField(const Detail::RFieldBase &field, int level)
+{
+   for (int i = 0; i < level; ++i) fOutput << "  ";
+   fOutput << "\"" << field.GetName() << "\": ";
+   
+   // A custom object (has ENTupleStructure::kRecord) should let its subfields display
+   // its entries instead of displaying all by itself.
+   // So the field of the custom object should only display its name and make an early quit here.
+   if (field.GetStructure() == ENTupleStructure::kRecord) {
+      fOutput << std::endl;
+      return;
+   }
+   
+   // const_cast may lead to undefined behaviour, so this procedure of obtaining
+   // the entries is only used when there are no good alternatives.
+   Detail::RFieldBase* fieldPtr = const_cast<Detail::RFieldBase*>(&field);
+   Detail::RFieldValue value{fieldPtr->GenerateValue()};
+   fieldPtr->Read(fIndex, &value);
+   fOutput << gInterpreter->ToString(field.GetType().c_str(), value.GetRawPtr());
```

Comment:
Oh, here we need the interpreter.  Or do we?  We could return a string saying "no support for <typename>"